### ISSUE TITLE
Negative length recorded in Chunk table

### ISSUE THREAD

**Issue by tosiara** [hashtopolis/server#712]
- Version: `0.12.0 commit c3fdbd1 branch master`

I have noticed negative length numbers recorded in the database:

```
mysql> select * from Chunk where taskid = 20;
+---------+--------+---------------+----------+---------+--------------+------------+---------------+----------+-------+---------+------------+
| chunkId | taskId | skip          | length   | agentId | dispatchTime | solveTime  | checkpoint    | progress | state | cracked | speed      |
+---------+--------+---------------+----------+---------+--------------+------------+---------------+----------+-------+---------+------------+
..
|  466270 |     20 | 5545364126763 |  8847360 |       7 |   1622094178 | 1622094353 | 5545372974123 |    10000 |     9 |       0 |          0 |
|  466271 |     20 | 5545367748654 | -3621891 |      15 |   1622628461 |          0 | 5545367748654 |        0 |     0 |       0 |          0 |
|  466272 |     20 | 5545372974123 |   188590 |      15 |   1622098353 | 1622098353 | 5545373162713 |    10000 |     4 |       0 |          0 |
| ..
|  466286 |     20 | 5545387135194 |   277501 |      15 |   1622107880 | 1622107891 | 5545387412695 |    10000 |     4 |       0 |          0 |
|  466287 |     20 | 5545390068957 | -2933763 |       6 |   1622465490 |          0 | 5545390068957 |        0 |     0 |       0 |          0 |
|  466288 |     20 | 5545400464158 |  8886450 |    NULL |            0 |          0 | 5545400464158 |        0 |     0 |       0 |          0 |
|  466289 |     20 | 5545397558649 |  2905509 |    NULL |            0 |          0 | 5545397558649 |        0 |     0 |       0 |          0 |
|  466290 |     20 | 5545387412695 |   277501 |       8 |   1622125058 | 1622125077 | 5545387690196 |    10000 |     4 |       0 |          0 |
|  466291 |     20 | 5545387690196 |  9868453 |      15 |   1622125096 | 1622125201 | 5545397558649 |    10000 |     4 |       0 |          0 |
+---------+--------+---------------+----------+---------+--------------+------------+---------------+----------+-------+---------+------------+
295972 rows in set (0.94 sec)
```
I don't know how exactly this happened. My system experienced random instability while working on a large task.
Maybe it is worth changing `length` column from `BIGINT` to `BIGINT UNSIGNED` as length is never expected to be negative?

**Comment by s3inlc** [comment]
That's interesting.. the skip value still was increased for the next chunks, even when having a negative length value. 

Changing the type of the column to unsigned could prevent it to happen and reporting errors instead, so that may be a good solution to better track where it exactly happens. I will implement the changes into the initial SQL file for the installation, but I will not apply the change via the update scripts as I suspect it could cause issues when upgrading (there needs just to be one negative number and the upgrade will error).

**Comment by zyronix** [comment]
Fixed in 2be8530862d0b91a049b01c361f6753f73778030
